api/envoy/api/v2/discovery.proto

```diff
@@ -176,7 +177,7 @@ message DeltaDiscoveryRequest {
   // When the DeltaDiscoveryRequest is a ACK or NACK message in response
   // to a previous DeltaDiscoveryResponse, the response_nonce must be the
   // nonce in the DeltaDiscoveryResponse.
-  // Otherwise response_nonce must be omitted.
+  // Otherwise (unlike in DiscoveryRequest) response_nonce must be omitted.
```

Comment:
What are the possibility of `otherwise`? 
First request on the wire is an obvious one. What else?

Reply:
When the `DeltaDiscoveryRequest` is being used as an actual request, i.e., "here are names of resources I am newly (un)interested in". This is getting into what I believe @htuch has mentioned, that these (Delta)DiscoveryRequest messages are a mix of two functionalities that would perhaps be better as explicitly separate messages: informing the server of Envoy's interest in resources, and ACKing server messages. With such a separation, this sort of rule wouldn't even need to be mentioned.

Reply:
Yeah, in UDPA we will make this separation. I think the clarifying comments are helpful in v2 incremental xDS, thanks @fredlas.